services/QuillLMS/app/controllers/admin_diagnostic_reports_controller.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+class AdminDiagnosticReportsController < ApplicationController
```

Comment:
This is 95% just a copy of `SnapshotsController` with tweaks to actions and `WORKERS_FOR_ACTIONS`.  I plan to follow up this PR with one that refactors this shared logic into a parent controller that both this and `SnapshotsController` inherit from.